osu.Game/Rulesets/Edit/Checks/CheckDelayedHitsounds.cs

```diff
@@ -0,0 +1,182 @@
+                        var points = waveform.GetPoints();
+
+                        // Skip muted samples
+                        if (points.Length == 0 || points.Sum(getAverageAmplitude) <= silence_threshold)
+                            continue;
+
+                        float maxAmplitude = points.Select(getAverageAmplitude).Max();
+
+                        int consequentDelay = 0;
+                        int delay = 0;
+                        float amplitude = 0;
+
+                        while (delay + consequentDelay < points.Length)
+                        {
+                            amplitude += getAverageAmplitude(points[delay]);
+
+                            // Reached peak amplitude/transient
+                            if (amplitude >= maxAmplitude)
+                                break;
+
+                            amplitude *= falloff_factor;
+
+                            if (amplitude < silence_threshold)
+                            {
+                                amplitude = 0;
+                                consequentDelay++;
+                            }
+
+                            delay++;
+                        }
+
+                        if (consequentDelay >= delay_threshold)
+                            yield return new IssueTemplateConsequentDelay(this).Create(file.Filename, consequentDelay);
+                        else if (consequentDelay + delay >= delay_threshold)
+                        {
+                            if (consequentDelay > 0)
+                                yield return new IssueTemplateDelay(this).Create(file.Filename, consequentDelay, delay);
+                            else
+                                yield return new IssueTemplateDelayNoSilence(this).Create(file.Filename, delay);
+                        }
+                        else if (consequentDelay + delay >= delay_threshold_negligible)
+                        {
+                            if (consequentDelay > 0)
+                                yield return new IssueTemplateMinorDelay(this).Create(file.Filename, consequentDelay, delay);
+                            else
+                                yield return new IssueTemplateMinorDelayNoSilence(this).Create(file.Filename, delay);
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool isHitSound(string filename)
+        {
+            if (!AudioCheckUtils.HasAudioExtension(filename))
+                return false;
+
+            // <bank>-<sampleset>
+            string[] parts = filename.ToLowerInvariant().Split('-');
+
+            if (parts.Length != 2)
+                return false;
+
+            string bank = parts[0];
+            string sampleSet = parts[1];
+
+            return HitSampleInfo.AllBanks.Contains(bank)
+                   && HitSampleInfo.AllAdditions.Concat(new[] { HitSampleInfo.HIT_NORMAL, HitSampleInfo.HIT_FLOURISH }).Any(sampleSet.StartsWith);
```

Comment:
I can understand appending `HIT_NORMAL`, but flourish? Not sure about that... Why?

Reply:
I think I'm just assuming things honestly, removed.